lib/FusionInventory/Agent/Task/Inventory/Generic/Dmidecode/Battery.pm

```diff
@@ -18,33 +18,40 @@ sub doInventory {
     my $inventory = $params{inventory};
     my $logger    = $params{logger};
 
-    my $battery = _getBattery(logger => $logger);
+    my @batteries = _getBatteries(logger => $logger);
```

Comment:
As _getBatteries() returns one array ref, you should avoid any confusion and use `$batteries` in place of `@batteries` in that  function. This will also fix the `return unless @batteries` condition which is actually always true: an array with one undef element is true as it contains one element.